python/main.py

```diff
@@ -0,0 +1,93 @@
+import functools
+import os
+
+
+def is_in_docker_container():
+    mp_in_container = os.getenv('METAPARTICLE_IN_CONTAINER', None)
+    if mp_in_container in ['true', '1']:
+        return True
+    elif mp_in_container in ['false', '0']:
+        return False
+
+    try:
+        with open('/proc/1/cgroup', 'r+t') as f:
+            lines = f.read().splitlines()
+            last_line = lines[-1]
+            if 'docker' in last_line:
+                return True
+            elif 'kubepods' in last_line:
+                return True
+            else:
+                return False
+
+    except FileNotFoundError:
+        return False
+
+
+def write_dockerfile(name):
+    with open('Dockerfile', 'w+t') as f:
+        f.write("""FROM python:3
+
+COPY ./ /{}/
+RUN pip install -r /{}/requirements.txt
+
+CMD python /{}/main.py
+    """.format(name, name, name))
+
+
+def select_builder(buildSpec):
+    if buildSpec == 'docker':
+        return __import__('docker_builder')
+    else:
+        raise Exception('Unknown buildSpec {}'.format(buildSpec))
+
+
+def select_runner(runSpec):
+    if runSpec == 'docker':
+        return __import__('docker_runner')
+    else:
+        raise Exception('Unknown buildSpec {}'.format(buildSpec))
+
+
+def get_name(options):
+    try:
+        return options['name']
+    except AttributeError:
+        pass
+
+    workdir = os.getcwd().split('/')
```

Comment:
Is there a reason you wouldn't want to use `os.path.basename(os.getwd())` as a default here?

Reply:
Great call, didn't realize you could do that. I'll update.